### DISCUSSION TITLE
how to get logs of rustdesk

### DISCUSSION THREAD

**Discussion by teqoit** [rustdesk/rustdesk#5999] [General]
how we can monitor logs of rustdesk access (self host). who is using rustdesk access and taking remote session through self host server

**Comment by DevJupyHUB** [comment]
We do this and it works well...

**ELK Stack** on remote log server + **Filebeat** (ligtweight log analysis) on RustDesk Server - with **GROK** pattern(s) / custom pre-processor + **Kibana** Dashboard.

(You may have trouble with the grok, since the "log format" is unique (no pre-built processor like for syslog) and you have to figure it out yourself line by line, but it will be a nice learning curve in the ELK world)

I'll help you get started :)

_**f.e.: **relaysever simpleMSG****_

[2023-02-28 20:33:27.580502 +02:00] INFO [src/relay_server.rs:441] Both are raw

**_in GROK:_** 

%{TIMESTAMP_ISO8601:timestamp} %{ISO8601_TIMEZONE:timezone}] %{LOGLEVEL:logLevel} \[%{DATA:source}\:%{NUMBER:sourceline}\] %{GREEDYDATA:message}


+++edit:
**_f.e.  IP change_after pre-proc:_**

![image](https://github.com/rustdesk/rustdesk/assets/125738232/18d2f256-0860-4499-bc27-47abf69fe768)

**Reply by teqoit** [reply]
thanks for the reply. 

i am looking for who is accessing / using rustdesk using my privatekey / server details. or active connections. can we get the logs.

**Comment by DevJupyHUB** [comment]
@teqoit _"can we get the logs."_

for sure...
you can find everything in the rustdesk related log files you just have to filter them out... =grok (regex)
(on Ubuntu /var/log/) 

f.e.: (in ELK env.)
https://www.elastic.co/guide/en/logstash/current/plugins-filters-grok.html

**Reply by adamency** [reply]
> "you can find everything in the rustdesk related log files"

No you cannot.

I have scoured all log files in `~/.local/share/logs/Rustdesk/*/` and `hbbs` `hbbr` and `rustdesk --server` and `rustdesk --service` logs. There is absolutely no reliable way to determine the number of active connections at a given time, nor when a connection started or stopped.

For instance, two connections from two different devices will only appear as one if they are close enough in time.

Rustdesk OSS is extremely unreliable for monitoring purposes, and the actual monitoring features are paywalled in the "pro" version.

**Reply by Jaynornj** [reply]
Thank you!

**Comment by Jaynornj** [comment]
Did you get it working? @teqoit

**Reply by hueldoeu** [reply]
https://github.com/rustdesk/rustdesk-server/discussions/470
